specifyweb/frontend/js_src/lib/components/Errors/JsonError.tsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import { backEndText } from '../../localization/backEnd';
+import { jsonStringify } from '../../utils/utils';
+import { className } from '../Atoms/className';
+import { TableIcon } from '../Molecules/TableIcon';
+
+type JsonResponse = {
+  exception: string;
+  message: string;
+  data: any;
+  formattedData: string;
+  traceback: string;
+};
+
+function createJsonResponse(error: string): JsonResponse {
+  const json = JSON.parse(error);
+  const hasLocalizationKey =
+    json.data.localizationKey == undefined ? false : true;
+  const jsonResponse = {
+    exception: json.exception,
+    message: hasLocalizationKey
+      ? resolveBackendLocalization(json)
+      : json.message,
+    data: json.data,
+    formattedData: jsonStringify(json.data, 2),
+    traceback: json.traceback,
+  };
+  return jsonResponse;
+}
+
+export function formatJsonBackendResponse(error: string): JSX.Element {
+  const response = createJsonResponse(error);
+  if (response.exception == 'BusinessRuleException')
+    return formatBusinessRuleException(error);
+  else if (response.exception == 'TreeBusinessRuleException')
+    return formatTreeBusinessRuleException(error);
+  else return formatBasicResponse(error);
+}
+
+function JsonBackendResponseFooter({
+  response,
+  isDataOpen = true,
+}: {
+  readonly response: JsonResponse;
+  readonly isDataOpen?: boolean;
+}): JSX.Element {
+  const hasData = response.data == null ? false : true;
+  return (
+    <>
+      {hasData && (
+        <details open={isDataOpen}>
+          <summary>Data</summary>
+          <pre>{response.formattedData}</pre>
+        </details>
+      )}
+      <details>
+        <summary>Traceback</summary>
+        <pre>{response.traceback}</pre>
+      </details>
+    </>
+  );
+}
+
+function BusinessRuleExceptionHeader({
+  table,
+  response,

```

Comment:
(totally optional)
Could replace `response,` with `response: {data, exception, message}`

Reply:
Is this needed (I can include it if need be)? Right below this I define response to be of type JsonResponse.

Reply:
Response type shouldn't be affected by this change. Full example:

```ts
{
  table,
  response: {data, exception, message}
}: {
  readonly table: string;
  readonly response JsonResponse;
}

Reply:
Ah ok! I see what you mean now. Did not know you could do that!

Reply:
Fixed in c1ee332